core/src/main/java/org/apache/accumulo/core/file/FileSKVIterator.java

```diff
@@ -33,6 +34,16 @@ public interface FileSKVIterator extends InterruptibleIterator, AutoCloseable {
 
   DataInputStream getMetaStore(String name) throws IOException, NoSuchMetaStoreException;
 
+  /**
+   * Returns an estimate of the number of entries that overlap the given extent. This is an estimate
+   * because the extent may or may not entirely overlap with each of the index entries included in
+   * the count. Will never underestimate but may overestimate.
+   *
+   * @param extent the key extent
+   * @return the estimate
+   */
+  long estimateEntries(KeyExtent extent) throws IOException;
```

Comment:
Would including the word `overlap` or `overlapping` in the method name make it more self-evident? Maybe something like `estimateOverlappingEntries`?